README.md
# Fancy Button Flutter ( Button with Icon )

Fancy Button - Fancy Button Flutter with Icon. Create awesome button with combination of Icon and Text. It is flexible and you can design according to your need.

![image description](https://github.com/dheeraj-bhadoria/Fancy-Button-Flutter-Button-With-Icon/blob/main/demo.png)

## Platform Support

| Android | iOS | MacOS | Web | Linux | Windows |
| :-----: | :-: | :---: | :-: | :---: | :-----: |
|   ✔️    | ✔️  |  ✔️   | ✔️  |  ✔️   |   ✔️    |


## Getting Started


With Flutter:

```yaml

$ flutter pub add fancy_button_flutter

```

This will add a line like this to your package's pubspec.yaml (and run an implicit flutter pub get):

```yaml

  dependencies:
  fancy_button_flutter: ^1.0.3+1

```

Alternatively, your editor might support flutter pub get. Check the docs for your editor to learn more.

## Example

First import the class in your class

```dart

import 'package:fancy_button_flutter/fancy_button_flutter.dart';

```

Example

```dart

    import 'package:fancy_button_flutter/fancy_button_flutter.dart';


    // This widget is the root of your application.
    @override 
    Widget build(BuildContext context) {
      /// Fancy button with its property 
      return FancyButton(
          button_icon: Icons.facebook,
          button_text: "  Login  ",
          button_height: 40,
          button_width: 150,
          button_radius: 50,
          button_color: Colors.blue,
          button_outline_color: Colors.blue,
          button_outline_width: 1,
          button_text_color: Colors.white,
          button_icon_color: Colors.white,
          icon_size: 22,
          button_text_size: 15,
          onClick: (){
            print("Button clicked");
          });
    }
    
```


## License

MIT License

Copyright (c) 2022 Dheeraj Singh Bhadoria

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

CHANGELOG.md
## [1.0.2+1] 

* Minor updates 

linux/my_application.cc
#include "my_application.h"

#include <flutter_linux/flutter_linux.h>
#ifdef GDK_WINDOWING_X11
#include <gdk/gdkx.h>
#endif

#include "flutter/generated_plugin_registrant.h"

struct _MyApplication {
  GtkApplication parent_instance;
  char** dart_entrypoint_arguments;
};

G_DEFINE_TYPE(MyApplication, my_application, GTK_TYPE_APPLICATION)

// Implements GApplication::activate.
static void my_application_activate(GApplication* application) {
  MyApplication* self = MY_APPLICATION(application);
  GtkWindow* window =
      GTK_WINDOW(gtk_application_window_new(GTK_APPLICATION(application)));

  // Use a header bar when running in GNOME as this is the common style used
  // by applications and is the setup most users will be using (e.g. Ubuntu
  // desktop).
  // If running on X and not using GNOME then just use a traditional title bar
  // in case the window manager does more exotic layout, e.g. tiling.
  // If running on Wayland assume the header bar will work (may need changing
  // if future cases occur).
  gboolean use_header_bar = TRUE;
#ifdef GDK_WINDOWING_X11
  GdkScreen* screen = gtk_window_get_screen(window);
  if (GDK_IS_X11_SCREEN(screen)) {
    const gchar* wm_name = gdk_x11_screen_get_window_manager_name(screen);
    if (g_strcmp0(wm_name, "GNOME Shell") != 0) {
      use_header_bar = FALSE;
    }
  }
#endif
  if (use_header_bar) {
    GtkHeaderBar* header_bar = GTK_HEADER_BAR(gtk_header_bar_new());
    gtk_widget_show(GTK_WIDGET(header_bar));
    gtk_header_bar_set_title(header_bar, "fancybuttonflutter");
    gtk_header_bar_set_show_close_button(header_bar, TRUE);
    gtk_window_set_titlebar(window, GTK_WIDGET(header_bar));
  } else {
    gtk_window_set_title(window, "fancybuttonflutter");
  }

  gtk_window_set_default_size(window, 1280, 720);
  gtk_widget_show(GTK_WIDGET(window));

  g_autoptr(FlDartProject) project = fl_dart_project_new();
  fl_dart_project_set_dart_entrypoint_arguments(project, self->dart_entrypoint_arguments);

  FlView* view = fl_view_new(project);
  gtk_widget_show(GTK_WIDGET(view));
  gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));

  fl_register_plugins(FL_PLUGIN_REGISTRY(view));

  gtk_widget_grab_focus(GTK_WIDGET(view));
}

// Implements GApplication::local_command_line.
static gboolean my_application_local_command_line(GApplication* application, gchar*** arguments, int* exit_status) {
  MyApplication* self = MY_APPLICATION(application);
  // Strip out the first argument as it is the binary name.
  self->dart_entrypoint_arguments = g_strdupv(*arguments + 1);

  g_autoptr(GError) error = nullptr;
  if (!g_application_register(application, nullptr, &error)) {
     g_warning("Failed to register: %s", error->message);
     *exit_status = 1;
     return TRUE;
  }

  g_application_activate(application);
  *exit_status = 0;

  return TRUE;
}

// Implements GObject::dispose.
static void my_application_dispose(GObject* object) {
  MyApplication* self = MY_APPLICATION(object);
  g_clear_pointer(&self->dart_entrypoint_arguments, g_strfreev);
  G_OBJECT_CLASS(my_application_parent_class)->dispose(object);
}

static void my_application_class_init(MyApplicationClass* klass) {
  G_APPLICATION_CLASS(klass)->activate = my_application_activate;
  G_APPLICATION_CLASS(klass)->local_command_line = my_application_local_command_line;
  G_OBJECT_CLASS(klass)->dispose = my_application_dispose;
}

static void my_application_init(MyApplication* self) {}

MyApplication* my_application_new() {
  return MY_APPLICATION(g_object_new(my_application_get_type(),
                                     "application-id", APPLICATION_ID,
                                     "flags", G_APPLICATION_NON_UNIQUE,
                                     nullptr));
}


windows/runner/main.cpp
#include <flutter/dart_project.h>
#include <flutter/flutter_view_controller.h>
#include <windows.h>

#include "flutter_window.h"
#include "utils.h"

int APIENTRY wWinMain(_In_ HINSTANCE instance, _In_opt_ HINSTANCE prev,
                      _In_ wchar_t *command_line, _In_ int show_command) {
  // Attach to console when present (e.g., 'flutter run') or create a
  // new console when running with a debugger.
  if (!::AttachConsole(ATTACH_PARENT_PROCESS) && ::IsDebuggerPresent()) {
    CreateAndAttachConsole();
  }

  // Initialize COM, so that it is available for use in the library and/or
  // plugins.
  ::CoInitializeEx(nullptr, COINIT_APARTMENTTHREADED);

  flutter::DartProject project(L"data");

  std::vector<std::string> command_line_arguments =
      GetCommandLineArguments();

  project.set_dart_entrypoint_arguments(std::move(command_line_arguments));

  FlutterWindow window(project);
  Win32Window::Point origin(10, 10);
  Win32Window::Size size(1280, 720);
  if (!window.CreateAndShow(L"fancybuttonflutter", origin, size)) {
    return EXIT_FAILURE;
  }
  window.SetQuitOnClose(true);

  ::MSG msg;
  while (::GetMessage(&msg, nullptr, 0, 0)) {
    ::TranslateMessage(&msg);
    ::DispatchMessage(&msg);
  }

  ::CoUninitialize();
  return EXIT_SUCCESS;
}


lib/fancy_button_flutter.dart
import 'package:flutter/material.dart';

/// Fancy button is a special button it provide you various property by which
/// you can create awsone button
/// You can set the icon, radius with in button
class FancyButton extends StatefulWidget {

  /// button icon to add icon in button
  IconData? button_icon = null;
  /// button text, It will show text in button
  String? button_text = "Button Text";
  /// button height
  double? button_height = 50;
  /// button width
  double? button_width = 100;
  /// to set the radius of the button you can use this property
  double? button_radius = 12;
  /// button color
  Color? button_color = Colors.black;
  /// button outline borer color
  Color? button_outline_color = Colors.white;
  /// button outline width you can set by it
  double? button_outline_width = 1;
  /// button text color
  Color? button_text_color = Colors.white;
  /// button icon color you can set by it
  Color? button_icon_color = Colors.white;
  /// you can also set the size of icon in button by button_size
  double? icon_size = 22;
  /// button text size property define the size of text in button
  double? button_text_size = 20;

  final Function() onClick;

  FancyButton({
    super.key,
    this.button_icon,
    this.button_text,
    this.button_height,
    this.button_width,
    this.button_radius,
    this.button_color,
    this.button_outline_color,
    this.button_outline_width,
    this.button_text_color,
    this.button_icon_color,
    this.icon_size,
    this.button_text_size,
    required this.onClick,
  });


  @override
  State<FancyButton> createState() => _FancyButtonState();
}

class _FancyButtonState extends State<FancyButton> {

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return InkWell(
        child: Container(
          decoration: BoxDecoration(
              color: widget.button_color ?? Colors.white,
              border: Border.all(
                color: widget.button_outline_color ?? Colors.black,
                width: widget.button_outline_width ?? 1,
              ),
              borderRadius: BorderRadius.all(Radius.circular(widget.button_radius ?? 12))
          ),
          width: widget.button_width ?? 100,
          height: widget.button_height ?? 40,
          child: Align(
            alignment: Alignment.center,
            child: Row(
              crossAxisAlignment: CrossAxisAlignment.center,
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                (widget.button_icon==null)?SizedBox():Icon(
                  widget.button_icon,
                  color: widget.button_icon_color ?? Colors.black,
                  size: widget.icon_size ?? 10,
                ),
                Text(
                  textAlign: TextAlign.center,
                  widget.button_text ?? "Button Text",
                  style: TextStyle(
                    color: widget.button_text_color ?? Colors.black,
                    fontSize: widget.button_text_size ?? 15,
                  ),
                )
              ],
            ),
          ),
        ),
        onTap: (){
            widget.onClick();
        },
    );
  }
}


example/main.dart
import 'package:fancy_button_flutter/fancy_button_flutter.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Column(
        children: [
          Row(
            children: [
              FancyButton(
                  button_text: "Button",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 0,
                  button_color: Colors.red,
                  button_outline_color: Colors.yellow,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
              ,
              SizedBox(width: 10,),
              FancyButton(
                  button_text: "Button",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 50,
                  button_color: Colors.red,
                  button_outline_color: Colors.yellow,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),
              SizedBox(width: 10,),
              FancyButton(
                  button_icon: Icons.account_circle_rounded,
                  button_text: "Button",
                  button_height: 40,
                  button_width: 150,
                  button_radius: 50,
                  button_color: Colors.red,
                  button_outline_color: Colors.yellow,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
            ],
          ),
          SizedBox(height: 10,),
          Row(
            children: [
              FancyButton(
                  button_text: "  Login  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 0,
                  button_color: Colors.blue,
                  button_outline_color: Colors.blue,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),

              SizedBox(width: 10,),

              FancyButton(
                  button_text: "  Login  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 50,
                  button_color: Colors.blue,
                  button_outline_color: Colors.blue,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
              ,
              SizedBox(width: 10,),

              FancyButton(
                  button_icon: Icons.facebook,
                  button_text: "  Login  ",
                  button_height: 40,
                  button_width: 150,
                  button_radius: 50,
                  button_color: Colors.blue,
                  button_outline_color: Colors.blue,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
            ],
          ),
          SizedBox(height: 10,),
          Row(
            children: [
              FancyButton(
                  button_text: "  Logout  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 0,
                  button_color: Colors.black,
                  button_outline_color: Colors.black,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),

              SizedBox(width: 10,),

              FancyButton(
                  button_text: "  Logout  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 50,
                  button_color: Colors.black,
                  button_outline_color: Colors.black,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),

              SizedBox(width: 10,),

              FancyButton(
                  button_icon: Icons.logout,
                  button_text: "  Logout  ",
                  button_height: 40,
                  button_width: 150,
                  button_radius: 50,
                  button_color: Colors.black,
                  button_outline_color: Colors.black,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
            ],
          ),
          SizedBox(height: 10,),
          Row(
            children: [
              FancyButton(
                  button_text: "  Chart  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 0,
                  button_color: Colors.deepOrange,
                  button_outline_color: Colors.deepOrange,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),
              SizedBox(width: 10,),

              FancyButton(
                  button_text: "  Chart  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 50,
                  button_color: Colors.deepOrange,
                  button_outline_color: Colors.deepOrange,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),
              SizedBox(width: 10,),

              FancyButton(
                  button_icon: Icons.add_chart,
                  button_text: "  Chart  ",
                  button_height: 40,
                  button_width: 150,
                  button_radius: 50,
                  button_color: Colors.deepOrange,
                  button_outline_color: Colors.deepOrange,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
            ],
          ),
          SizedBox(height: 10,),
          Row(
            children: [
              FancyButton(
                  button_text: "Add",
                  button_height: 50,
                  button_width: 100,
                  button_radius: 0,
                  button_color: Colors.green,
                  button_outline_color: Colors.green,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),
              SizedBox(width: 10,),

              FancyButton(
                  button_text: "  Add  ",
                  button_height: 40,
                  button_width: 100,
                  button_radius: 50,
                  button_color: Colors.green,
                  button_outline_color: Colors.green,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  }),

              SizedBox(width: 10,),

              FancyButton(
                  button_icon: Icons.add_circle_outline,
                  button_text: "  Add  ",
                  button_height: 40,
                  button_width: 150,
                  button_radius: 50,
                  button_color: Colors.green,
                  button_outline_color: Colors.green,
                  button_outline_width: 1,
                  button_text_color: Colors.white,
                  button_icon_color: Colors.white,
                  icon_size: 22,
                  button_text_size: 15,
                  onClick: (){
                    print("Button clicked");
                  })
            ],
          ),
        ],
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
